torch/ao/sparsity/sparsifier.py

```diff
@@ -0,0 +1,167 @@
+            self.config = []
+            stack = [model]
+            while stack:
+                module = stack.pop()
+                for name, child in module.named_children():
+                    if type(child) in SUPPORTED_MODULES:
+                        self.config.append(child)
+                    else:
+                        stack.append(child)
+
+        for module_config in self.config:
+            if isinstance(module_config, nn.Module):
+                module_config = {'module': module_config}
+            local_args = copy.deepcopy(self.defaults)
+            local_args.update(module_config)
+            module = local_args['module']
+            module_path = _module_to_path(self.model, module)
+            if module_path and module_path[0] == '.':
+                module_path = module_path[1:]
+            local_args['path'] = module_path
+            self.module_groups.append(local_args)
+
+    def __getstate__(self):
+        return {
+            'defaults': self.defaults,
+            'state': self.state,
+            'module_groups': self.module_groups,
+        }
+
+    def __setstate__(self, state):
+        self.__dict__.update(state)
+
+    def __repr__(self):
+        format_string = self.__class__.__name__ + ' ('
+        for i, sparse_args in enumerate(self.module_groups):
+            module = sparse_args['module']
+            format_string += '\n'
+            format_string += f'\tModule Group {i}\n'
+            format_string += f'\t    module: {module}\n'
+            for key in sorted(sparse_args.keys()):
+                if key == 'module':
+                    continue
+                format_string += f'\t    {key}: {sparse_args[key]}\n'
+        format_string += ')'
+        return format_string
+
+    def state_dict(self):
+        r"""Returns the state of the optimizer as a :class:`dict`.
+
+        It contains:
+        * module_groups - a list containing all sparsity configuration groups
+            with the key 'path' specifying the layer path within a model
+        """
+        module_groups = [
+            dict(filter(lambda key_value: key_value[0] != 'module', mg.items()))
+            for mg in self.module_groups
+        ]
+        return {
+            'module_groups': module_groups,
+        }
+
+    def load_state_dict(self, state_dict, strict=True):
+        module_groups = copy.deepcopy(state_dict['module_groups'])
+        for group in module_groups:
+            layer = _path_to_module(self.model, group['path'])
+            if strict and layer is None:
+                raise RuntimeError(f'Error loading group["path"] into the model')
+            group['module'] = layer
+        self.__setstate__({'module_groups': module_groups})
+
+    def prepare(self, *args, **kwargs):
+        r"""Adds mask parametrization to the layer weight
+        """
+        for config in self.module_groups:
+            module = config['module']
+            if not getattr(module, 'mask', None):
```

Comment:
Lets replace this with fakesparse module insertions, which should contain the mask